lib/pbench/agent/tool_meister_start.py

```diff
@@ -0,0 +1,1366 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""pbench-tool-meister-start - start the execution of the Tool Meister
+sub-system.
+
+There are two roles tool-meister-start plays:
+
+  1. (optionally) orchestrate the creation of the instances of the Redis
+     server, Tool Data Sink, and Tool Meisters
+
+  2. execute the start sequence for the Tool Meister sub-system
+
+The `--orchestrate=(create|existing)` command line parameter is used to control
+the orchestration.  The default is to "create" the instances of the Redis
+server, Tool Data Sink, and Tool Meisters.  If the user specifies "existing",
+then the command assumes the `--redis-server` and `--tool-data-sink` parameters
+will be provided to direct the command to the location of those instances.
+
+The sequence of steps to execute the above behaviors is as follows:
+
+   1. Loading tool group data for the requested tool group
+      - This is the first step regardless so that the tool information can be
+        validated, the number of TMs and their marching orders can be
+        enumerated, and when orchestrating the TMs, the list of where those
+        TMs are requested to run can be determined
+   2. [orchestrate] Starting a Redis server
+   3. Creating the Redis channel for the TDS to talk to the client
+      - <prefix>-to-client
+      - TDS publishes, a client subscribes
+      - This is how the TDS reports back to a client the success or failure
+        of requested actions
+   4. Pushing the loaded tool group data and metadata into the Redis server
+      for the TDS and all the TMs
+   5. [orchestrate] Starting the local Tool Data Sink process
+   6. [orchestrate] Starting all the local and remote Tool Meisters
+   7. Waiting for the TDS to send a message reporting that it, and all the TMs,
+      started
+      - The TDS knows all the TMs that were started from the registered tools
+        data structure argument given to it
+   8. Verify all the requested Tool Meisters have reported back and that their
+      tool installation checks were successful
+
+There is a specific flow of data between these various components.  This		￼
```

Comment:
Looks like an extra line was added.

Reply:
Of course, there had to be something. Weird, though.

Reply:
This line, incidentally, came along with trailing whitespace that nobody complained about. I thought black didn't like that?

Reply:
Most of the tools are pretty tolerant of whitespace (particularly trailing) inside block comments.